Purchases/Misc/Purchases+async.swift

```diff
@@ -102,19 +100,14 @@ extension Purchases {
                 guard let customerInfo = maybeCustomerInfo else {
                     fatalError("Expected non-nil result 'customerInfo' for nil error")
                 }
-                guard let transaction = maybeTransaction else {
```

Comment:
This made cancelling purchases crash.

Reply:
Should maybe keep this guard but add `!userCancelled`? It seems like that was maybe its original intention 🤔 

```swift
guard let transaction = maybeTransaction, !userCancelled else {
```

And the same to the ones below this too

Reply:
But we do have a transaction when the user cancelled, so why not send it?